### ISSUE TITLE
Acts::Result fails for bool return type

### ISSUE THREAD

**Issue by paulgessinger** [acts-project/acts-core-test#347]
[GitLab issue: #591](https://gitlab.cern.ch/acts/acts-core/issues/591)

Result<> seems not to be working for bools.
I get the following error message:

```
constructor not viable: no known conversion from 'bool' to 'std::variant<bool, error_code> &&'
for 1st argument
  Result(std::variant<T, E>&& var) : m_var(var) {}
```

**Comment by paulgessinger** [comment]
[Original comment by: pagessin at 2019-04-17 08:05:41](https://gitlab.cern.ch/acts/acts-core/issues/591#note_2531803)

 I want to implement a fix for this before we tag 0.9.0

**Comment by paulgessinger** [comment]
[Original comment by: pagessin at 2019-04-17 09:26:19](https://gitlab.cern.ch/acts/acts-core/issues/591#note_2532514)

 The problem here is that std::error_code is auto-convertible to bool. That means the implicit conversion from `std::error_code` to `Result<bool, std::error_code` is ambiguous, since the compiler cannot figure out whether you want to construct the ok state or the error state. If I removed this restriction, constructing this type from `std::error_code` would actually construct the `bool` ok state, since it's automatically converted.

To remedy this, you will have to use `Result::success(true|false)` to construct the ok value and `Result::failure(std::error_code)` to construct the error value.
